components/tool/taskManager/src/index.js

```diff
@@ -12,16 +12,31 @@ import useBeforeUnloadEffect from './hooks/useBeforeUnloadEffect.js'
 import useContext from './hooks/useContext.js'
 import useDevMode from './hooks/useDevMode.js'
 
-export default function ToolTaskManager({isVisible = true, statusIcons = {}}) {
+export default function ToolTaskManager({
+  isVisible = true,
+  statusIcons = {},
+  taskIdsFilter = [],
+  taskAmountFilter = 0
+}) {
   const {countWork, countFinishedWork, getState} = useContext()
   const state = getState()
   const {isDevModeEnabled, registerClick} = useDevMode()
   useBeforeUnloadEffect({isVisible})
   const getItems = () => {
     return (
       <>
-        {state.tasks.map(task => {
+        {state.tasks.map((task, index) => {
           if (task.visibleWork === 0 && isDevModeEnabled === false) return
+
+          if (
+            Array.isArray(taskIdsFilter) &&
+            taskIdsFilter.length > 0 &&
+            taskIdsFilter.find(id => id === task.id) === undefined
```

Comment:
```suggestion
            Array.isArray(taskIdsFilter) &&
            taskIdsFilter.some(id => id === task.id)
```

would this work for you and perhaps simplify the logic? 🤔

Reply:
At first we had the statement that way, but we prefered to not filter tasks at all if the array was empty, to keep things simple by default.